types/src/dkg.rs

```diff
@@ -54,3 +74,27 @@ impl OnChainConfig for DKGState {
     const MODULE_IDENTIFIER: &'static str = "dkg";
     const TYPE_IDENTIFIER: &'static str = "DKGState";
 }
+
+pub trait DKGTrait {
+    type PrivateParams;
+    type PublicParams: Send + Sync;
+    type Transcript: Clone + Default + Send + Sync + for<'a> Deserialize<'a>;
+
+    fn generate_transcript<R: CryptoRng>(
+        rng: &mut R,
+        sk: &Self::PrivateParams,
+        params: &Self::PublicParams,
+    ) -> Self::Transcript;
+
+    fn verify_transcript(params: &Self::PublicParams, trx: &Self::Transcript) -> Result<()>;
+
+    fn aggregate_transcripts(
+        params: &Self::PublicParams,
+        base: &mut Self::Transcript,
+        extra: &Self::Transcript,
+    );
+}
+
+pub trait DKGPrivateParamsProvider<DKG: DKGTrait> {
```

Comment:
Do you really need to make this a trait?

Reply:
It will may be  `IdentityBlob` in production, and the exact private key in tests.
the reason why it is `IdentityBlob` in production is that the private key in it is not Clonable.

Reply:
would it be simpler to just implement From for them?

Reply:
you mean `From<IdentityBlob> for RealDKGPrivateParams` for example?
IMO it doesn't help much since IdentityBlob is a black box, the exact key is part of it but i can't take it out.